### ISSUE TITLE
Add a _changeBlock attribute on all entities

### ISSUE THREAD

**Issue by azf20** [graphprotocol/graph-node#2958]

**Comment by Jannis** [comment]
Closely related: https://github.com/graphprotocol/graph-node/issues/295 (confirmations API). The confirmations API _might_ be overkill but it's worth considering use cases before we add an API change that we then need to maintain forever.

**Comment by azf20** [comment]
Yes there are definitely overlaps, though I think the confirmations API requires quite a lot more functionality (field level details, details on the changes, finalization) which is beyond the scope of this issue.

You do raise a good point on `an API change that we then need to maintain forever`. I think there is a question whether we want to maintain the `_meta` pattern that exists at the top level of the subgraph, where `_changeBlock` is part of `_meta` (and which leaves more scope for potential extension in the future).

**Comment by dotansimha** [comment]
Does this a field a consumer might need as data, or metadata? (needed for app development, or while querying/testing the subgraph API?)

Because I think we can consider creating a spec for `thegraph-introspection` that exposes those? Just like GraphQL has it's own introspection that starts with `__`, we can introduce an internal introspection that can be part of every `type` / `schem`, and have it's own lifecycle? (so we can decide if we break/maintain it or not, and make a statement that it's an internal API that is subject to changes?)

**Comment by azf20** [comment]
Interesting - if that is an established pattern that could work, and as you say it has the benefit of flexibility. To answer the first question, this would be useful for app development, for example when polling for new data, or finding recently changed entities - so that does make it a bit more difficult to break anything, once introduced

**Comment by azf20** [comment]
OK. Revisiting this, I am tempted to opt for the simpler approach described in this issue: https://github.com/graphprotocol/graph-node/issues/1838
This adds this functionality purely as a new filter in the top-level `where`, rather than as an extension to the entity itself. This means that it is not available to Select, or Orderby.

**Comment by azf20** [comment]
Closing for now as this was implemented as a filter, we can establish whether adding this information to the entity would be useful
